src/Couchbase.Lite.Shared/API/Sync/Replicator.cs

```diff
@@ -458,42 +473,53 @@ private bool HandleError(C4Error error)
             return true;
         }
 
-        private void OnDocEnded(List<ReplicatedDocument> replications, bool pushing)
+        private void OnDocEnded(ConcurrentBag<ReplicatedDocument> replications, bool pushing)
         {
             if (_disposed) {
                 return;
             }
 
-            for(int i = 0; i < replications.Count; i++) {
+            while (!replications.IsEmpty) {
+                _conflictTasks.Add(Task.Run(() =>
+                {
+                    ReplicatedDocument replication;
+                    if (replications.TryTake(out replication)) {
+                        try {
+                            Config.Database.ResolveConflict(replication.Id, Config.ConflictResolver);
+                            replication = replication.ClearError();
+                        } catch (Exception e) {
+                            C4Error error;
+                            if(e.GetBaseException() is CouchbaseException) {
```

Comment:
Only CouchbaseLiteException will have C4 error codes.  The rest of the Couchbase exception family has different domain codes.